### PR TITLE
New event. CommandProcessEvent

### PR DESCRIPTION
**Pull request by djxy**
I needed an event to handle the process of a command before and after that happen in the same tick. So I created this new event to handle that.

When some commands are fired by some players, I need to disable some functionalities. /help is one of them.

Currently there is only one event for when a command is send and it is SendCommandEvent. The commandResult isn't good.

https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/command/SpongeCommandManager.java#L258

SpongeAPI: https://github.com/SpongePowered/SpongeAPI/pull/1333

### CODE REVIEW

File: src/main/java/org/spongepowered/common/command/SpongeCommandManager.java [hunk 1]
```diff
@@ -24,29 +24,19 @@
...
 import org.spongepowered.api.Sponge;
-import org.spongepowered.api.command.CommandCallable;
-import org.spongepowered.api.command.CommandException;
-import org.spongepowered.api.command.CommandManager;
-import org.spongepowered.api.command.CommandMapping;
-import org.spongepowered.api.command.CommandPermissionException;
-import org.spongepowered.api.command.CommandResult;
-import org.spongepowered.api.command.CommandSource;
-import org.spongepowered.api.command.InvocationCommandException;
+import org.spongepowered.api.command.*;
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/command/SpongeCommandManager.java]: No star imports please

File: src/main/java/org/spongepowered/common/command/SpongeCommandManager.java [hunk 2]
```diff
@@ -57,23 +47,17 @@
...
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Function;
 
-import javax.activation.CommandMap;
-import javax.annotation.Nullable;
-import javax.inject.Inject;
+import static com.google.common.base.Preconditions.checkNotNull;
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/command/SpongeCommandManager.java]: Static imports should remain at the top

File: src/main/java/org/spongepowered/common/command/SpongeCommandManager.java (hunk 1 above)
Comment (djxy) [inline src/main/java/org/spongepowered/common/command/SpongeCommandManager.java]: My bad, Intellij Idea did it automatically. I will change that.

File: src/main/java/org/spongepowered/common/command/SpongeCommandManager.java (hunk 2 above)
Comment (djxy) [inline src/main/java/org/spongepowered/common/command/SpongeCommandManager.java]: Intellij Idea... I will look to disable that.

**Comment by zml2008** [comment]
I don't see any reason to have SendCommandEvent and CommandProcessEvent.Pre be separate -- their functionality is close enough that you should be able to combine them

**Comment by djxy** [comment]
Yes, I talked with @ryantheleach and we thought their functionalities were close. I will combine them.

**Comment by djxy** [comment]
I removed SendCommandEvent and I replaced it by CommandProcessEvent.Pre

**Pull request closed without merging**